ruby-challenges.rb

```diff
@@ -19,16 +27,60 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+def total array
+    array.sum
+    end
+p total nums_array1
+p total nums_array2
 
-
-# --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
+# --------------------3 a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. 
+#Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
-
-
-
-# -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
+class Bike
+    def initialize model, wheels, current_speed
+        @model = model
+        @wheel = wheels
+        @current_speed = current_speed
+    end
+    def bike_info
+       p "The #{@model} bike has #{@wheel} wheels and is going #{@current_speed} mph"
+    end
+    
+# -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase 
+    #the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
 
 # Expected output example: my_bike.pedal_faster(10) => 10
 # Expected output example: my_bike.pedal_faster(10) => 20
 # Expected output example: my_bike.brake(25) => 0
+def pedal_faster current_speed
+  @current_speed += current_speed
+end
+def brake current_speed
+  if current_speed > @current_speed 
+    @current_speed=0
+  end
+  if @current_speed > 0
+    @current_speed -=current_speed
+  end
+end
+def pedal_faster current_speed
+  @current_speed += current_speed
+end
+def brake current_speed
+  if current_speed >= @current_speed 
+    return 0
+  end
+  if @current_speed > 0
+    @current_speed -=current_speed
+  end
+end
+end
```

Comment:
Excellent job on the ruby code! I know it's difficult to move back and fourth but you did a great job!